ostd/src/mm/page_table/mod.rs

```diff
@@ -380,4 +380,22 @@ pub trait PageTableEntryTrait:
     /// The level of the page table the entry resides is given since architectures
     /// like amd64 only uses a huge bit in intermediate levels.
     fn is_last(&self, level: PagingLevel) -> bool;
+
+    /// Reads the value from the given pointer using an atomic read operation.
+    ///
+    /// # Safety
+    ///
+    /// - `ptr` must be [valid] for reading the `Self` type value.
+    ///
+    /// [valid]: crate::mm::io#safety
+    unsafe fn atomic_read(ptr: *const Self, order: Ordering) -> Self;
```

Comment:
I suggest `read_atomic`/`write_atomic`, which should be more consistent with `read_volatile`/`write_volatile` (and other `read_xxx`/`write_xxx` methods) in the Rust standard library.